### PR TITLE
(PE-27489) Add ability to specify ttl

### PR DESCRIPTION
**Pull request by tvpartytonight**
This change adds in the ability to specify a ttl for the generate and
sign actions.

### CODE REVIEW

File: lib/puppetserver/ca/action/generate.rb [hunk 1]
```diff
@@ -232,7 +235,7 @@ def generate_certs(certnames, alt_names, settings, digest)
               @logger.inform "Certificate for #{certname} was autosigned."
               true
```
Comment (Magisus) [inline lib/puppetserver/ca/action/generate.rb]: Can we add something in here (a warning maybe?) that when the cert is autosigned, the TTL will be ignored?

File: lib/puppetserver/ca/certificate_authority.rb [hunk 2]
```diff
@@ -35,11 +45,40 @@ def make_ca_url(resource_type = nil, certname = nil)
...
+        match = /^(\d+)(s|m|h|d|y)?$/.match(ttl)
+        if match
+          if match[2]
+            match[1].to_i * UNITMAP[match[2]].to_i
+          else
+            ttl
+          end
+        else
+          @logger.err "Error:"
+          @logger.err " '{ttl}' is an invalid ttl value"
```
Comment (Magisus) [inline lib/puppetserver/ca/certificate_authority.rb]: I think this needs a '#' on it

**Comment by Magisus** [comment]
I also manually tested this, and it seems to work great!

File: lib/puppetserver/ca/action/generate.rb [hunk 3]
```diff
@@ -75,6 +75,9 @@ def self.parser(parsed = {})
                     'Causes the cert to be generated offline.') do |ca_client|
               parsed['ca-client'] = true
             end
+            opts.on('--ttl TTL', 'the time-to-live for each cert generated and signed') do |ttl|
```
Comment (justinstoller) [inline lib/puppetserver/ca/action/generate.rb]: All of the other help texts start with a capital letter.

File: lib/puppetserver/ca/action/sign.rb [hunk 4]
```diff
@@ -32,6 +32,9 @@ class Sign
         def self.parser(parsed = {})
           OptionParser.new do |opts|
             opts.banner = BANNER
+            opts.on('--ttl TTL', 'the time-to-live for each cert signed') do |ttl|
```
Comment (justinstoller) [inline lib/puppetserver/ca/action/sign.rb]: ditto

File: lib/puppetserver/ca/action/generate.rb [hunk 5]
```diff
@@ -230,9 +233,12 @@ def generate_certs(certnames, alt_names, settings, digest)
             # Check if the CA autosigned the cert
             if download_cert(ca, certname, settings)
```
Comment (justinstoller) [inline lib/puppetserver/ca/action/generate.rb]: could we move this into a separate method? the containing method is already big and getting bigger, this conditional seems like a good chunk to split out??

**Pull request merged**